gateway/gateway_test.go

```diff
@@ -1322,28 +1321,32 @@ func (stp *testSetup) snSend(pkt snPkts.Packet, setMsgID bool) {
 		}
 	}
 
-	err := pkt.Write(stp.snConn)
+	buf, err := pkt.Pack()
+	if err != nil {
+		stp.t.Fatal(err)
+	}
+	_, err = stp.snConn.Write(buf)
 	if err != nil {
 		stp.t.Fatal(err)
 	}
 }
 
 func (stp *testSetup) snRecv() snPkts.Packet {
-	buff := make([]byte, maxTestPktLength)
-	n, err := stp.snConn.Read(buff)
+	rawPacket := make([]byte, maxTestPktLength)
+	n, err := stp.snConn.Read(rawPacket)
```

Comment:
Little bit non-clear readable n,h,p ... still code will be compiled, so better tu use some self-descriptive names of variables.

Reply:
@mprymek Take a look pls :) non-blocking comment

Reply:
For record: we have agreed we will discuss particular uses lately.